src/main/java/com/parsing/parsers/pdf/parsing/DataRecognizer.java

```diff
@@ -0,0 +1,156 @@
+                        amounts.add(findAmount(row.getColumns().get(x).getParsingResult()));
+                    }
+                    if (findPrices(row.getColumns().get(x).getParsingResult()) != null) {
+                        prices.add(findPrices(row.getColumns().get(x).getParsingResult()));
+                    }
+                }
+                getAmount(amounts);
+                getPriceAndTotalPrice(prices);
+                result = saveItems();
+            }
+        }
+        return result;
+    }
+
+    private boolean saveItems() {
+        if (totalPrice != null && totalPrice.equals(BigDecimal.valueOf(amount).multiply(price))) {
+            lotPDFResultService.saveLaptopItem(model, price, amount);
+            return true;
+        }
+        return false;
+    }
+
+    private void getAmount(List<String> amounts) {
+        if (amounts.size() == 1) {
+            amount = Integer.parseInt(amounts.get(0));
+        } else if (amounts.size() > 1) {
+            log.debug("more than one amount number found in one row");
+            amount = 0;
+        }
+        if (amount == 0 && price != null) {
+            amount = totalPrice.divide(price).toBigInteger().intValueExact();
+        }
+    }
+
+    private void getPriceAndTotalPrice(List<String> prices) {
+        List<BigDecimal> bigDecimals = null;
+        try {
+            bigDecimals = prices.stream().map(BigDecimal::new).toList();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        Optional<BigDecimal> min = bigDecimals.stream().min(Comparator.naturalOrder());
+        Optional<BigDecimal> max = bigDecimals.stream().max(Comparator.naturalOrder());
+        if (min.isPresent() && max.isPresent()) {
+            price = min.get();
+            totalPrice = max.get();
+        }
+    }
+
+    private int getModelColumnNumber(Row row) {
+        int modelColumnNumber = 0;
+        for (Column column : row.getColumns()) {
+            if (isModel(column.getParsingResult())) {
+                model = findModel(column.getParsingResult());
+                modelColumnNumber = row.getColumns().indexOf(column);
+            }
+        }
+        return modelColumnNumber;
+    }
+
+    private boolean isModelRow(Row row) {
+        List<String> listLowerCaseColumn = row.getColumns().stream().map(x -> x.getParsingResult().toLowerCase(Locale.ROOT)).toList();
+        for (String column : listLowerCaseColumn) {
+            for (String laptopModel : laptopModels) {
+                if (column.contains(laptopModel)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private String findModel(String input) {
+        for (String laptop : laptopModels) {
+            if (input.toLowerCase(Locale.ROOT).contains(laptop)) {
+                return input;
+            }
+        }
+        return null;
+    }
+
+    private boolean isModel(String input) {
```

Comment:
findModel & isModel looks like almost same method. if in your opinion you is nesseseri to have isModel method, realization could by in 1 line -> { return findModel(input) != null;}

Reply:
Yeah, looks like it.

Reply:
Done.